core/src/test/java/com/linecorp/armeria/server/logging/AccessLoggerIntegrationTest.java

```diff
@@ -59,6 +59,7 @@ void testAccessLogger() throws Exception {
         assertThat(server.blockingWebClient().get("/").status().code()).isEqualTo(200);
         assertThat(server.requestContextCaptor().size()).isEqualTo(1);
         final ServiceRequestContext ctx = server.requestContextCaptor().poll();
-        assertThat(CTX_REF).hasValue(ctx);
+        assertThat(ctx).isNotNull();
+        assertThat(CTX_REF).hasValueSatisfying(ctxRef -> assertThat(ctxRef.id()).isEqualTo(ctx.id()));
```

Comment:
Otherwise, should we use `equalsIgnoreWrapper()` instead?

Reply:
@ikhoon 's comment actually made me realize that the problem may not be equality since we never wrap `ServiceRequestContext` unless we enable context leak detection (which we aren't).

I think this is probably more of a timing issue, so I've added an awaitility check instead.
(so the atomic reference is updated after comparison fails, but before the message is constructed)
https://github.com/assertj/assertj/blob/b26b28106a72d0962e833877622983d6fafe82f4/assertj-core/src/main/java/org/assertj/core/api/AtomicReferenceAssert.java#L51-L52